File: src/TheWorld/Api/TripController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Threading.Tasks;
using AutoMapper;
using Microsoft.AspNet.Authorization;
using Microsoft.AspNet.Mvc;
using TheWorld.Models;
using TheWorld.ViewModels;

// For more information on enabling MVC for empty projects, visit http://go.microsoft.com/fwlink/?LinkID=397860

namespace TheWorld.Api
{
    [Authorize]
    [Route("api/trips")]
    public class TripController : Controller
    {
        private readonly IWorldRepository _worldRepository;

        public TripController(IWorldRepository worldRepository)
        {
            _worldRepository = worldRepository;
        }
        [HttpGet("")]
        public JsonResult Get()
        {
            var results = _worldRepository.GetUserTripsWithStops(User.Identity.Name);
            var rt = Mapper.Map<IEnumerable<TripViewModel>>(results);
            return Json(rt);
        }
        [HttpPost("")]
        public JsonResult Post([FromBody]TripViewModel trip)
        {

            try
            {
                if (ModelState.IsValid)
                {

                    var newTrip = Mapper.Map<Trip>(trip);
                    newTrip.UserName = User.Identity.Name;
                    //Save to the database
                    _worldRepository.AddTrip(newTrip);
                    if (_worldRepository.SaveAll())
                    {
                        Response.StatusCode = (int)HttpStatusCode.Created;
                        return Json(Mapper.Map<TripViewModel>(newTrip));
                    }
                }
            }
            catch (Exception e)
            {
                Response.StatusCode = (int)HttpStatusCode.BadRequest;
                return Json(new { e.Message });
            }
            Response.StatusCode = (int)HttpStatusCode.BadRequest;
            return Json(new { Message = "Failed", ModelState = ModelState });
        }



    }
}

File: src/TheWorld/wwwroot/js/tripController.js
(function () {
    "use strict";
    angular.module("app-trips")
        .controller("tripsController", tripsController);

    function tripsController($http) {

        var vm = this;

        vm.trips = [];

        vm.newTrip = {};

        vm.errorMessage = "";

        vm.isBusy = true;

        $http.get("/api/trips")
            .then(function (response) {
                //Success
                angular.copy(response.data, vm.trips);

            }, function (error) {
                //Failure
                vm.errorMessage = "Failed to load to data : " + error;
            }).finally(function () {
                vm.isBusy = false;
            });

        vm.addTrip = function () {
            //vm.trips.push({ name: vm.newTrip.name, created: new Date() });
            //vm.newTrip= {}
            vm.isBusy = true;
            $http.post("/api/trips", vm.newTrip)
                .then(function (response) {
                    //success
                    vm.trips.push(response.data);
                    vm.newTrip = {};
                }, function (error) {
                    vm.errorMessage = "Failed to save new trip" + error;
                })
                .finally(function () {
                    vm.isBusy = false;
                });
        };
    }
})();